Sources/Utility/POSIXConvenience.swift

```diff
@@ -11,11 +11,9 @@
 import POSIX
 
 public func fopen(path: String..., mode: FopenMode = .Read, body: (UnsafeMutablePointer<FILE>) throws -> Void) throws {
-    var fp = try POSIX.fopen(Path.join(path), mode: mode)
-    defer { if fp != nil { fclose(fp) } }
+    let fp = try POSIX.fopen(Path.join(path), mode: mode)
+    defer { fclose(fp) }
     try body(fp)
-    fclose(fp)  // defer is not necessarily immediate
```

Comment:
I really don't think that should happen…or rather, I can't think of any way that would happen that *wouldn't* do the same thing with the manual invocation.